libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -22,16 +22,30 @@
  */
 
 #include <libsolidity/analysis/StaticAnalyzer.h>
-
 #include <libsolidity/analysis/ConstantEvaluator.h>
 #include <libsolidity/ast/AST.h>
+#include <libsolidity/ast/ASTUtils.h>
 #include <liblangutil/ErrorReporter.h>
+
+#include <range/v3/view/enumerate.hpp>
+
 #include <memory>
 
 using namespace solidity;
 using namespace solidity::langutil;
 using namespace solidity::frontend;
 
+namespace
```

Comment:
This is a copy of the same function from [`TypeChecker`](https://github.com/ethereum/solidity/blob/develop/libsolidity/analysis/TypeChecker.cpp#L83-L93) which I'd like to move into a common utils file. Would `ASTUtils` be appropriate for this? I'd do this in another PR of course.

Reply:
I did it anyway, it's better this way https://github.com/ethereum/solidity/pull/15148.